### PR TITLE
Storage bucket and folder delete of versioning objects.

### PR DESCRIPTION
**Pull request by AjayShivanagol**

### CODE REVIEW

File: packages/storage-backend/storage-lib/src/main/java/com/daimler/data/minio/client/DnaMinioClientImp.java [hunk 1]
```diff
@@ -365,8 +365,7 @@ public List<String> listObjectsInBucket(String userId, String bucketName) {
 
 				// Lists objects information.
 				LOGGER.debug("Listing Objects information from minio for user:{}", userId);
-				Iterable<Result<Item>> results = minioClient.listObjects(ListObjectsArgs.builder().bucket(bucketName).recursive(true).build());
-
+				Iterable<Result<Item>> results = minioClient.listObjects(ListObjectsArgs.builder().bucket(bucketName).includeVersions(true).recursive(true).build());
```
Comment (boonbaluri) [inline packages/storage-backend/storage-lib/src/main/java/com/daimler/data/minio/client/DnaMinioClientImp.java:368]: not required here

File: packages/storage-backend/storage-lib/src/main/java/com/daimler/data/minio/client/DnaMinioClientImp.java [hunk 2]
```diff
@@ -842,13 +841,19 @@ public MinioGenericResponse removeObjects(String userId, String bucketName, Stri
 					for (String path: prefixValue) {
 					    for (String objectName: objectNames) {
 							 if (objectName.contains(path)) {
-								objects.add(new DeleteObject(objectName));
+								 // Fetch the versions for this object
```
Comment (boonbaluri) [inline packages/storage-backend/storage-lib/src/main/java/com/daimler/data/minio/client/DnaMinioClientImp.java:844]: this loop for adding each version is not required

File: packages/storage-backend/storage-lib/src/main/java/com/daimler/data/minio/client/DnaMinioClientImp.java [hunk 3]
```diff
@@ -842,13 +841,19 @@ public MinioGenericResponse removeObjects(String userId, String bucketName, Stri
...
+								 }
 							 }
 					    }
                     }
                 }
 
-                LOGGER.info("Removing objects from Minio bucket:{}", bucketName);
+				LOGGER.info("Removing objects from Minio bucket:{}", bucketName);
 				Iterable<Result<DeleteError>> results = minioClient
 						.removeObjects(RemoveObjectsArgs.builder().bucket(bucketName).objects(objects).build());
```
Comment (boonbaluri) [inline packages/storage-backend/storage-lib/src/main/java/com/daimler/data/minio/client/DnaMinioClientImp.java:858]: .objects(objects).bypassGovernanceMode(true)

if this works, then only remove above code

File: packages/storage-backend/storage-lib/src/main/java/com/daimler/data/minio/client/DnaMinioClientImp.java [hunk 4]
```diff
@@ -898,30 +903,55 @@ public MinioGenericResponse removeBucket(String userId, String bucketName) {
...
+				LOGGER.debug("Fetch secret from vault successful for user:{}", userId);
 				MinioClient minioClient = MinioClient.builder().endpoint(minioBaseUri).credentials(userId, userSecretKey)
 						.build();
-				LOGGER.info("Removing bucket:{} from Minio", bucketName);
-				minioClient.removeBucket(RemoveBucketArgs.builder().bucket(bucketName).build());
-				LOGGER.info("Success from Minio remove Bucket:{}", bucketName);
-				
-				LOGGER.info("Removing policies for bucket:{}",bucketName);
-				List<String> policies = Arrays.asList(bucketName+"_"+ConstantsUtility.READ,bucketName+"_"+ConstantsUtility.READWRITE);
+
```
Comment (boonbaluri) [inline packages/storage-backend/storage-lib/src/main/java/com/daimler/data/minio/client/DnaMinioClientImp.java:909]: this change is not required

**Review by Namitha-01dev — APPROVED** [review]

**Pull request merged**